modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -871,9 +871,16 @@ def get_objects_from_partitions(cls, partitions):
             assert all(
                 [len(partition.list_of_blocks) == 1 for partition in partitions]
             ), "Implementation assumes that each partition contains a single block."
-            return cls._execution_wrapper.materialize(
-                [partition.list_of_blocks[0] for partition in partitions]
-            )
+            blocks_to_materialize = [
+                partition.list_of_blocks[0] for partition in partitions
+            ]
+            try:
+                result = cls._execution_wrapper.materialize(blocks_to_materialize)
+            except Exception:
+                # wait for the end of computations so that the processes free up the resources used
+                cls._execution_wrapper.wait(blocks_to_materialize)
+                raise
```

Comment:
I don't think this is exactly the right place to put this synchronization point... At least it should be waited conditionally based on some configuration variable, as I would think that in most scenarios the user doesn't want to wait for all computations when it's already known that an exception is happening.

Also, this is fixing a problem which isn't entirely related to the original issue, so maybe it's better to open a new separate PR with this change.

Reply:
> Also, this is fixing a problem which isn't entirely related to the original issue, so maybe it's better to open a new separate PR with this change.

I agree, let's continue in another PR.